Alex/12/AOC2021_12B_v00.py

```diff
@@ -67,88 +67,51 @@ def can_i_visit_this_small_cave(small_cave,path_count):
 			if cave.islower() is True and cave_occurances >= 2:
 				#Another cave has been visited twice. We cannot visit our small cave again.
 				visit_small_cave_bool = False
-		
-		if visit_small_cave_bool == '':
-			#no other small cave has occured twice. We may visit this one again.
-			visit_small_cave_bool = True
-		
-	else:
-		#Small cave has not been visited. You may visit it
-		visit_small_cave_bool = True
-	
-	
-	
 	return visit_small_cave_bool
 
 
-
-
-def produce_new_paths(path,nodesdict,paths,path_count):
+def produce_new_paths(path,nodesdict):
 	#Produce new paths from path
 	
-	new_paths = []
+	new_paths = set()
 	
 	connected_nodes = nodesdict[path[-1]]
 	
 	for node in connected_nodes:
 		if node == 'start':
 			continue
 		
-		if node.islower() is True:
+		if node.islower():
+			path_count = Counter(path)
 			visit_small_cave_bool = can_i_visit_this_small_cave(node,path_count)
 			
 			if visit_small_cave_bool == False:
 				continue
 		
-		new_path = path + [node]
-		
-		if new_path in paths:
-			continue
-		else:
-			new_paths.append(new_path)
+		new_path = *path, node

```

Comment:
'path' is now a tuple as lists aren't hashable, so can't be stored in sets. This is doing the equivalent of making a new list from a new element and an old list.